flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-maxcompute/src/main/java/org/apache/flink/cdc/connectors/maxcompute/utils/MaxComputeUtils.java

```diff
@@ -0,0 +1,258 @@
+        odps.setEndpoint(maxComputeOptions.getEndpoint());
+        odps.setTunnelEndpoint(maxComputeOptions.getTunnelEndpoint());
+        odps.setDefaultProject(maxComputeOptions.getProject());
+        odps.setUserAgent("Flink CDC");
+        return odps;
+    }
+
+    public static TableTunnel getTunnel(
+            MaxComputeOptions maxComputeOptions, MaxComputeWriteOptions writeOptions) {
+        Odps odps = getOdps(maxComputeOptions);
+        Configuration configuration =
+                Configuration.builder(odps)
+                        .withRetryLogger(RetryUtils.getRetryLogger())
+                        .withRetryPolicy(new RetryUtils.FlinkDefaultRetryPolicy())
+                        .withCompressOptions(
+                                MaxComputeUtils.compressOptionOf(
+                                        writeOptions.getCompressAlgorithm()))
+                        .withQuotaName(maxComputeOptions.getQuotaName())
+                        .build();
+        TableTunnel tunnel = new TableTunnel(odps, configuration);
+        if (!StringUtils.isNullOrWhitespaceOnly(maxComputeOptions.getTunnelEndpoint())) {
+            tunnel.setEndpoint(maxComputeOptions.getTunnelEndpoint());
+        }
+        return tunnel;
+    }
+
+    public static Table getTable(MaxComputeOptions maxComputeOptions, TableId tableId) {
+        Odps odps = getOdps(maxComputeOptions);
+        if (maxComputeOptions.isSupportSchema()) {
+            return odps.tables()
+                    .get(
+                            maxComputeOptions.getProject(),
+                            tableId.getNamespace(),
+                            tableId.getTableName());
+        } else {
+            return odps.tables().get(tableId.getTableName());
+        }
+    }
+
+    public static TableSchema getTableSchema(MaxComputeOptions options, TableId tableId) {
+        Odps odps = getOdps(options);
+        if (options.isSupportSchema()) {
+            return odps.tables()
+                    .get(options.getProject(), tableId.getNamespace(), tableId.getTableName())
+                    .getSchema();
+        } else {
+            return odps.tables().get(options.getProject(), tableId.getTableName()).getSchema();
+        }
+    }
+
+    public static boolean supportSchema(MaxComputeOptions maxComputeOptions) {
+        Odps odps = getOdps(maxComputeOptions);
+        try {
+            boolean flag =
+                    Boolean.parseBoolean(
+                            odps.projects().get().getProperty(Constant.SCHEMA_ENABLE_FLAG));
+            LOG.info("project {} is support schema: {}", maxComputeOptions.getProject(), flag);
+            return flag;
+        } catch (OdpsException e) {
+            throw new UncheckedOdpsException(e);
+        }
+    }
+
+    public static CompressOption compressOptionOf(String compressAlgo) {
+        CompressOption.CompressAlgorithm compressAlgorithm;
+        switch (compressAlgo) {
```

Comment:
What about using CompressOption.CompressAlgorithm.valueOf() to simplify this?

Reply:
Perhaps using zlib and lz4 instead of odps_zlib and odps_lz4_frame would be clearer for users?